### DISCUSSION TITLE
How to override lsp handlers correctly

### DISCUSSION THREAD

**Discussion by sidequestboy** [mason-org/mason.nvim#1511] [Q&A]
I use mason-lspconfig's automatic `setup_handlers` function to set up my LSP servers. That snippet looks like this:

```lua
mason_lspconfig.setup_handlers {
  function(server_name)
    require('lspconfig')[server_name].setup {
      capabilities = capabilities,
      on_attach = on_attach,
      settings = servers[server_name],
      filetypes = (servers[server_name] or {}).filetypes,
      handlers = {
        ['textDocument/hover'] = vim.lsp.with(vim.lsp.handlers.hover, { silent = true }),
      },
    }
  end,
}

```

The problem I am having is with that overridden hover handler - I am trying to pass silent = true to avoid the "No Information Available" message when an LSP doesn't have hover info ([see here](https://www.reddit.com/r/neovim/comments/1182s32/is_it_possible_to_disable_no_information/)), but with this override I instead get errors like the following when I call `vim.lsp.buf.hover`:

```
Error executing vim.schedule lua callback: ...atest/nvim-macos/share/nvim/runtime/lua/vim/lsp/util.lua:1784: 'width' key must be a positive Integer
stack traceback:
	[C]: in function 'nvim_open_win'
	...atest/nvim-macos/share/nvim/runtime/lua/vim/lsp/util.lua:1784: in function 'handler'
	...tly/latest/nvim-macos/share/nvim/runtime/lua/vim/lsp.lua:1505: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by williamboman** [comment]
I'd consult the lspconfig documentation. That function is only called by mason-lspconfig when it detects a server is installed, it doesn't intercept or interfere any interactions with lspconfig, so for LSP client issues you'd wanna check lspconfig or `:h lsp`.
